misc/config_tools/data/generic_board/generic_code/hybrid/misc_cfg.h

```diff
@@ -7,16 +7,16 @@
 #ifndef MISC_CFG_H
 #define MISC_CFG_H
 
-#define SOS_ROOTFS "root=/dev/sda3"
-#define SOS_CONSOLE "console=ttyS0"
+#define SOS_ROOTFS "root=/dev/sda3 "
```

Comment:
How do you think about adding in both place?
I actually thought I should even add whitespace at in the beginning of the SOS_ROOTFS, SOS_CONSOLE, SOS_ROOTFS in case of any re-ordering or re-usage of the macros...

SOS_VM_BOOTARGS is generated with :
 <xsl:template name ="sos_vm_bootarges">
    <xsl:if test="count(vm[vm_type='SOS_VM'])">
      <xsl:value-of select="acrn:comment(concat('SOS_VM == VM', vm[vm_type='SOS_VM']/@id))" />
      <xsl:value-of select="$newline" />
      <xsl:value-of select="acrn:define('SOS_VM_BOOTARGS', 'SOS_ROOTFS SOS_CONSOLE SOS_IDLE SOS_BOOTARGS_DIFF', '')" />
    </xsl:if>
  </xsl:template>

To achieve your suggestion, the line 
<xsl:value-of select="acrn:define('SOS_VM_BOOTARGS', 'SOS_ROOTFS SOS_CONSOLE SOS_IDLE SOS_BOOTARGS_DIFF', '')" /> 
will become something like:
<xsl:variable name=padding select="concat($quot, $whitespace, $quot)"/>
<xsl:value-of select="acrn:define('SOS_VM_BOOTARGS', concat('SOS_ROOTFS',  $padding, 'SOS_CONSOLE', $padding, 'SOS_IDLE', $padding, 'SOS_BOOTARGS_DIFF'))" />

that looks fine with me, but in case somehow they are referred in other place, maybe we should add double (the end of macros and the place concatenate the macros) even triple (the end of macros and the place concatenate the macros and the beginning of the macros) safeguard?

Reply:
In my experience, it's odd to see a trailing space on the defined string, particularly without a comment saying why the extra space is there, otherwise it looks more like a typo. It seems safer to define the strings without any assumptions. Any code that uses strings like these should assume responsibility for concatenating them with any needed separation characters. 

Any recommendations from the other reviewers?

Reply:
@NanlinXie @terryzouhao @wenlingz @szhen11